domain.tf

```
@@ -1,19 +1,26 @@
+# Locals configuration to check if we should create SES validation or not (allow external management of this)
+locals {
+  enable_ses_validation = var.domain_validation ? 1 : 0
+}
+
 data "aws_route53_zone" "default" {
   name = var.domain
 }
 
 resource "aws_route53_record" "default" {
+  count   = local.enable_ses_validation
   zone_id = data.aws_route53_zone.default.zone_id
-  name    = "_amazonses.${aws_ses_domain_identity.default.id}"
+  name    = "_amazonses.${aws_ses_domain_identity.default.0.id}"
```

usually we use a join splat for each of these

could you change each one?

```suggestion
  name    = "_amazonses.${join("", aws_ses_domain_identity.default.*.id)}"
```